source/atomic_ops.cpp

```diff
@@ -63,7 +62,56 @@ bool atomic_cas<uint32_t>(uint32_t *ptr, uint32_t *expectedCurrentValue, uint32_
 
     return !__STREXW(desiredValue, ptr);
 }
-#endif /* #if (__CORTEX_M >= 0x03) */
+
+uint32_t atomic_incr(uint32_t *valuePtr, uint32_t delta)
+{
+    uint32_t newValue;
+    do {
+        newValue = __LDREXW(valuePtr) + delta;
```

Comment:
I understand that this could be implemented with ```atomic_cas``` but I don't understand why it needs to be.  This implementation looks simpler, faster, and easier to understand to me. I agree that, where ```__LDREX()``` and ```__STREX()``` are unavailable, it is a requirement to use ```atomic_cas``` to implement ```atomic_incr``` and ```atomic_decr```, but I can't see the same requirement for implementations where ```__LDREX()``` and ```__STREX()``` are available. Is there something I've missed?